cscs-checks/tools/profiling_and_debugging/ipm_mpi_omp.py

```diff
@@ -0,0 +1,85 @@
+import os
+
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+@rfm.required_version('>=2.14')
+@rfm.parameterized_test(['C'], ['C++'], ['F90'])
+class Ipm(rfm.RegressionTest):
+    def __init__(self, lang):
+        super().__init__()
+        self.name = 'IPM_%s' % lang.replace('+', 'p')
+        self.descr = '%s check' % lang
+        self.language = lang
+        self.valid_systems = ['daint:gpu', 'daint:mc', 'dom:gpu', 'dom:mc']
+        self.valid_prog_environs = ['PrgEnv-cray', 'PrgEnv-gnu',
+                                    'PrgEnv-intel', 'PrgEnv-pgi']
+        self.prgenv_flags = {
+            'PrgEnv-gnu': ['-O2 -g -fopenmp'],
+            'PrgEnv-cray': ['-O2 -g -homp'],
+            'PrgEnv-intel': ['-O2 -g -openmp'],
+            'PrgEnv-pgi': ['-O2 -g -mp']
+        }
+        self.ipm_modules = {
+            'PrgEnv-gnu': 'IPM/2.0.6-CrayGNU-18.08',
```

Comment:
Enclose the module names in brackets `[]` so that you can use directly `self.modules = self.ipm_modules[environ.name]` on line 63.